### PR TITLE
ci(runner): use ubuntu 24.04 runner, ensure podman 4 from repo installed, use localstack:stable

### PR DESCRIPTION
**Pull request by andrewazores**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits using a GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification)

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

See https://github.com/cryostatio/cryostat/pull/577#issuecomment-2245782283

## Description of the change:
*This change allows an environment variable to be configured so that...*

## Motivation for the change:
*This change is helpful because users may want to...*

## How to manually test:
1. *Run CRYOSTAT_IMAGE=quay.io... bash smoketest.bash...*
2. *...*

### CODE REVIEW

**Comment by andrewazores** [comment]
testcontainers is still starting to fail the `localstack:stable` here for some reason...

**Comment by andrewazores** [comment]
It starts up fine locally, so maybe something about the CI environment or the CI runner image changed?

**Comment by mwangggg** [comment]
` Ign:25 https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04  InRelease
Ign:25 https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04  InRelease
Ign:25 https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04  InRelease
Err:25 https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04  InRelease
  503  Service Unavailable [IP: 195.135.223.226 443]` 
found this in the install podman step

**Comment by andrewazores** [comment]
Ah, that could make sense too. It's running with an old podman version and for whatever reason that breaks the health check detection?

**Comment by andrewazores** [comment]
Logs do indicate that because the [OpenSUSE mirror is down](https://status.opensuse.org/), CI just ended up reinstalling podman 3.x even though we want to explicitly install v4.

I'll update this PR to also ensure the CI installs the podman package from the expected OpenSUSE repository, so that we don't get silent failures like this if there's another outage in the future.

**Comment by andrewazores** [comment]
Aha, there is another way forward: Ubuntu 24.04 has podman 4 in its own native repositories, and since May GitHub has had 24.04 in public beta. Let's give that a try.

**Comment by andrewazores** [comment]
https://github.com/actions/runner-images/issues/9691

**Comment by andrewazores** [comment]
Test run: https://github.com/andrewazores/cryostat3/actions/runs/10064838666/job/27822648368

**Comment by mwangggg** [comment]
hmm seems like it's stuck installing packages

**Comment by andrewazores** [comment]
I don't think it's necessarily stuck, it's just trying to build the arm64 container image which it has to do by emulating an arm instruction set on top of the amd64 runner, so it's always very slow.

Recent successful builds take ~70 minutes: https://github.com/cryostatio/cryostat/actions/runs/10060318052

**Comment by andrewazores** [comment]
That change happened in https://github.com/cryostatio/cryostat/pull/373 , the CI time is significantly slower after that:

image

But, it's unfortunately the price to be paid to ensure that the native dependencies within the container are really the correct ones - at least with the current approach. The old fast building containers would contain the wrong native dependencies (ie netty: https://github.com/cryostatio/cryostat/blob/1929e79e758b818ab8208639019ff5cde39bb522/pom.xml#L505), so features like podman/docker container discovery that depend on Unix domain sockets (a netty native feature) would only actually work on amd64 hosts.

There's probably, hopefully, some other better way to make sure the right dependencies get into the multiarch containers and get the build to run faster again, but I haven't looked any further into it.

**Comment by andrewazores** [comment]
Test run succeeded.

**Comment by tthvo** [comment]
`ubuntu-24.04` runner image will likely be GA in August btw so we can use `ubuntu:latest` again :D https://github.com/actions/runner-images/issues/9691#issuecomment-2063926929

File: pom.xml [hunk 1]
```diff
@@ -54,7 +54,7 @@
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
     <quarkus.platform.version>3.2.12.Final</quarkus.platform.version>
-    <quarkus-quinoa.version>2.3.6</quarkus-quinoa.version>
+    <quarkus-quinoa.version>2.2.5</quarkus-quinoa.version>
```
Comment (andrewazores) [inline pom.xml:57]: https://github.com/cryostatio/cryostat/pull/581/files#r1690272578

**Comment by andrewazores** [comment]
Ready for review.

**Review by aali309 — COMMENTED** [review]
Looks good to me.

**Pull request merged**